flashcards/web/src/components/AddDecksForm.tsx

```diff
@@ -0,0 +1,154 @@
+import React from "react";
+import { useFormik } from "formik";
+import {
+    Button,
+    Checkbox,
+    Container,
+    FormControlLabel,
+    Grid,
+    InputLabel,
+    MenuItem,
+    Select,
+    TextField,
+    Typography,
+} from "@mui/material";
+import * as Yup from "yup";
+
+function AddDecksForm() {
+    const formik = useFormik({
+        validateOnChange: false,
+        initialValues: {
+            tagName: "",
+            author: "",
+            selectTag: "",
+            selectCategory: "",
+            selectAvailability: "",
+            price: "",
+            checkboxPublic: false,
+            checkboxPrice: false,
+        },
+        onSubmit: (values) => {
+            alert(JSON.stringify(values, null, 2));
+        },
+        validationSchema: Yup.object().shape({
+            tagName: Yup.string().min(2, "Too Short!").max(20, "Too Long!").required("Required!"),
+            author: Yup.string().min(2, "Too Short!").max(20, "Too Long!").required("Required!"),
+            checkboxPrice: Yup.boolean(),
+            checkboxPublic: Yup.boolean(),
+            price: Yup.number().when("checkboxPrice", {
+                is: true,
+                then: Yup.number().required("Required!"),
+            }),
+            selectTag: Yup.string().required("Required!"),
+            selectCategory: Yup.string().required("Required!"),
+        }),
+    });
+    return (
+        <form onSubmit={formik.handleSubmit}>
+            <Container maxWidth="md">
+                <Grid container>
+                    <Grid item xs={12}>
+                        <Typography>Add decks form</Typography>
+                    </Grid>
+                    <Grid item xs={12} marginBottom={2}>
+                        <TextField
+                            fullWidth
+                            name="tagName"
+                            label="Tag name"
+                            onChange={formik.handleChange}
+                            value={formik.values.tagName}
+                        />
+                        {formik.errors.tagName && <div className="text-red-600">{formik.errors.tagName}</div>}
```

Comment:
https://formik.org/docs/api/errormessage
replace it with ErrorMessage component (applies to all checks)

Reply:
Do I have to use Formik to use ErrorMessage?

Reply:
But you're using formal already

Reply:
I mean <Formik /> component, instead of useFormik. When I add ErrorMassage to the form, it crashes.

Reply:
Valid point, checked docs you're right, although same doc page strongly discourages using `useFormik` unless you're forced to do so.
Either keep the hook and use MaterialUI native error messages or use Formik component as context provider.
This version merges disadvantages of both 😄 

[useFormik](https://formik.org/docs/api/useFormik)
[materialUI](https://formik.org/docs/examples/with-material-ui)